### PR TITLE
[react] Replace Enzyme by @testing-library/react

### PR DESCRIPTION
**Pull request by qmonmert**
In order to prepare the passage to **React v17.0** 🚀, it will be a good solution to use **Testing Library** rather than **Enzyme** (it's an advice from Dan Abramov on Twitter).

Indeed Enzyme is not compatible with the first RC of **React v17.0**: 17.0.0-rc.0 ❌

With this PR, we prepare the passage to **React v17.0** ✅

[PR React 17](https://github.com/jhipster/generator-jhipster/pull/12241)  🛠

image

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/files-react.js [hunk 1]
```diff
@@ -313,7 +313,6 @@ const files = {
             path: TEST_SRC_DIR,
             templates: [
                 'jest.conf.js',
-                'spec/enzyme-setup.ts',
```
Comment (mshima) [inline generators/client/files-react.js:316]: Cleanup this file.

Comment (qmonmert) [reply generators/client/files-react.js:316]: I deleted this file

Comment (mshima) [reply generators/client/files-react.js:316]: Remove the file at:
https://github.com/jhipster/generator-jhipster/blob/03980190ab493f251166841a5bc1741e9b95af17/generators/cleanup.js#L130

**Comment by mshima** [comment]
Frontend test didn’t fail but there is a lot of errors printed.

File: generators/client/files-react.js (hunk 1 above)
Comment (qmonmert) [reply generators/client/files-react.js:316]: yes done

File: generators/client/templates/react/src/test/javascript/spec/app/shared/auth/private-route.spec.tsx.ejs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
-import { Route } from 'react-router-dom';
-import { shallow } from 'enzyme';
+import { Router } from 'react-router-dom';
+import { shallow } from 'enzyme'; import { createMemoryHistory } from 'history';
```
Comment (MathieuAA) [inline generators/client/templates/react/src/test/javascript/spec/app/shared/auth/private-route.spec.tsx.ejs:3]: you forgot to remove enzyme here (+ the new line)

Comment (qmonmert) [reply generators/client/templates/react/src/test/javascript/spec/app/shared/auth/private-route.spec.tsx.ejs:3]: yes I correct

**Comment by qmonmert** [comment]
@mshima do you think that you can merge this PR ?

**Comment by DanielFran** [comment]
@qmonmert merged!

**Comment by mshima** [comment]
Thanks @qmonmert.

**Pull request merged**